zebra/zebra_vty.c

```diff
@@ -603,66 +603,127 @@ static void vty_show_ip_route_detail(struct vty *vty, struct route_node *rn,
  */
 static void show_route_nexthop_helper(struct vty *vty,
 				      const struct route_entry *re,
-				      const struct nexthop *nexthop)
+				      const struct nexthop *nexthop,
+				      json_object *json)
 {
 	char buf[MPLS_LABEL_STRLEN];
 	int i;
 
 	switch (nexthop->type) {
 	case NEXTHOP_TYPE_IPV4:
 	case NEXTHOP_TYPE_IPV4_IFINDEX:
-		vty_out(vty, " via %pI4", &nexthop->gate.ipv4);
-		if (nexthop->ifindex)
-			vty_out(vty, ", %s",
-				ifindex2ifname(nexthop->ifindex,
-					       nexthop->vrf_id));
+		if (json)
+			json_object_string_addf(json, "ip", "%pI4",
+						&nexthop->gate.ipv4);
+		else
+			vty_out(vty, " via %pI4", &nexthop->gate.ipv4);
+		if (nexthop->ifindex) {
+			if (json)
+				json_object_string_add(
+					json, "interfaceName",
+					ifindex2ifname(nexthop->ifindex,
+						       nexthop->vrf_id));
+			else
+				vty_out(vty, ", %s",
+					ifindex2ifname(nexthop->ifindex,
+						       nexthop->vrf_id));
+		}
 		break;
 	case NEXTHOP_TYPE_IPV6:
 	case NEXTHOP_TYPE_IPV6_IFINDEX:
-		vty_out(vty, " via %s",
-			inet_ntop(AF_INET6, &nexthop->gate.ipv6, buf,
-				  sizeof(buf)));
-		if (nexthop->ifindex)
-			vty_out(vty, ", %s",
-				ifindex2ifname(nexthop->ifindex,
-					       nexthop->vrf_id));
+		if (json)
+			json_object_string_addf(json, "ip", "%pI6",
+						&nexthop->gate.ipv6);
+		else
+			vty_out(vty, " via %s",
+				inet_ntop(AF_INET6, &nexthop->gate.ipv6, buf,
+					  sizeof(buf)));
+		if (nexthop->ifindex) {
+			if (json)
+				json_object_string_add(
+					json, "interfaceName",
+					ifindex2ifname(nexthop->ifindex,
+						       nexthop->vrf_id));
+			else
+				vty_out(vty, ", %s",
+					ifindex2ifname(nexthop->ifindex,
+						       nexthop->vrf_id));
+		}
 		break;
 
 	case NEXTHOP_TYPE_IFINDEX:
-		vty_out(vty, " is directly connected, %s",
-			ifindex2ifname(nexthop->ifindex,
-				       nexthop->vrf_id));
+		if (json)
+			json_object_string_add(json, "directlyConnected",
+					       ifindex2ifname(nexthop->ifindex,
+							      nexthop->vrf_id));
+		else
+			vty_out(vty, " is directly connected, %s",
+				ifindex2ifname(nexthop->ifindex,
+					       nexthop->vrf_id));
 		break;
 	case NEXTHOP_TYPE_BLACKHOLE:
-		vty_out(vty, " unreachable");
+		if (!json)
+			vty_out(vty, " unreachable");
+		else
+			json_object_boolean_true_add(json, "unreachable");
 		switch (nexthop->bh_type) {
 		case BLACKHOLE_REJECT:
-			vty_out(vty, " (ICMP unreachable)");
+			if (json)
+				json_object_boolean_true_add(json, "reject");
+			else
+				vty_out(vty, " (ICMP unreachable)");
 			break;
 		case BLACKHOLE_ADMINPROHIB:
-			vty_out(vty, " (ICMP admin-prohibited)");
+			if (json)
+				json_object_boolean_true_add(
+					json, "admin-prohibited");
```

Comment:
`adminProhibited` (camelCase format is needed for JSON keys).

Reply:
Fixed it